File: go/ipse.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	recurse(`package main

import (
	"fmt"
	"strings"
)

func main() {`, `
}`, `
func recurse(A, B, R string) {
	fmt.Println(A)
`+"	fmt.Println(\"	recurse(`%s`, `%s`, `%s`), A, B, R)\")"+`
	fmt.Println(B)
	fmt.Println(R)
}`)
}

func recurse(A, B, R string) {
	fmt.Println(A)
	repl := strings.ReplaceAll(R, "\"", "\\\"")
	fmt.Printf("	recurse(`%s`, `%s`, `%s`)", A, B, repl)
	fmt.Println(B)
	fmt.Println(R)
}

File: quines/python/ipse.py
#!/usr/bin/env python3
def recurse(clear, shebang):
    print(shebang)
    print(clear)
    repl = clear.replace("\\", "\\\\")
    print(f"recurse(\"\"\"{repl}\"\"\", \"{shebang}\")")

recurse("""def recurse(clear, shebang):
    print(shebang)
    print(clear)
    repl = clear.replace("\\\\", "\\\\\\\\")
    print(f"recurse(\\"\\"\\"{repl}\\"\\"\\", \\"{shebang}\\")")
""", "#!/usr/bin/env python3")
